utils/utils.go

```diff
@@ -252,6 +258,30 @@ func unmarshalJSONFiles(path string, ignoreDirErr bool) ([]interface{}, error) {
 	return files, nil
 }
 
+func unmarshalJSONFilesWithFilenames(path string) (map[string]interface{}, error) {
+	fileInfos, err := ioutil.ReadDir(path)
+	if err != nil {
+		return map[string]interface{}{}, err
```

Comment:
could this just be `return nil, err` here with your changes to the unmarshaling logic on the backend?  (same question on the error handling for json.Unmarshal below)